frontend/src/host_orchestrator/orchestrator/instancemanager.go

```diff
@@ -105,6 +112,9 @@ func (m *CVDToolInstanceManager) CreateCVD(req apiv1.CreateCVDRequest) (apiv1.Op
 	if err := validateRequest(&req); err != nil {
 		return apiv1.Operation{}, operator.NewBadRequestError("invalid CreateCVDRequest", err)
 	}
+	if err := m.hostValidator.Validate(); err != nil {
```

Comment:
Is this check only done on `CreateCVD`? Whether the host is valid or not can be determined before a CVD is created. We can fail even faster if we add a new route to the host orchestrator that returns whether the host is valid or not (200-OK = valid, 503 = invalid). This endpoint could be checked as part of the `CreateHost` operation in the cloud orchestrator and fail it if appropriate.

The check should probably still happen on `CreateCVD` too, since it would avoid extra work in cases where the host was not created by the cloud orchestrator.

Reply:
yes, I was thinking on a new endpoint to get the relevant specs (nested virtualization enabled among them) of the host to know whether valid or not. still we need this check on the create cvd logic.